compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -2352,8 +2352,8 @@ public void visit(BLangTupleDestructure tupleDeStmt, AnalyzerData data) {
 
         setTypeOfVarRef(tupleDeStmt.varRef, data);
 
-        BType type = typeChecker.checkExpr(tupleDeStmt.expr, data.env, tupleDeStmt.varRef.getBType(), data.prevEnvs,
-                data.commonAnalyzerData);
+        BType type = typeChecker.checkExpr(tupleDeStmt.expr, data.env, symTable.noType, data.prevEnvs,
+                        data.commonAnalyzerData);
```

Comment:
As discussed during the call, this isn't quite correct.

The following results in a compilation error with this change.

```ballerina
public function main() {
    int a;
    float b;

    [a, b] = [1, 2];
}
```